components/nimbus/src/enrollment.rs

```diff
@@ -946,6 +947,12 @@ where
         .collect()
 }
 
+pub(crate) fn sort_experiments_by_published_date(experiments: &[Experiment]) -> Vec<Experiment> {
+    let mut experiments = experiments.to_vec();
+    experiments.sort_by(|a, b| a.published_date.cmp(&b.published_date));
```

Comment:
This clone is noted with caution. I think a comment would be good, so we can come back to it if/when it starts being a problem.